execute.go

```diff
@@ -149,37 +149,90 @@ loop:
 				goto loop
 			}
 		case opcall:
-			if backtrack {
-				break loop
-			}
```

Comment:
You cannot remove these 3 lines.

Reply:
Ok thanks. Think the reason i moved it into only be done for non-iterator opcall case was to try mimic how `opeach` case does things but i probably got it wrong. Will have to think more about it. Also don't like that there is some duplication the way I did this now.

Reply:
Oops, you're right.

Reply:
But i'm guessing the unit test fail might be because i do a pop too much in some error case, haven't look closer at it yet.